src/utils/reducers/ShoppingCartReducer.js

```diff
@@ -0,0 +1,71 @@
+export const ShoppingStateEnums = {
+  DELETE_ITEMS: 0,
+  MINUS_ITEM: 1,
+  ADD_ITEM: 2,
+  PLUS_ITEM: 3,
+};
+
+export const initialStateShoppingCart = {
+  count: 0,
+  cartInfo: []
+};
+
+const shoppinCartReducer = (state, action) => {
+  const { type, payload } = action;
+  let copyCartInfo = [...state.cartInfo];
+  const indexItem = state.cartInfo.findIndex(
+    (cartItem) => cartItem.id == payload.id
+  );
+
+  switch (type) {
+    case ShoppingStateEnums.DELETE_ITEMS:
+      copyCartInfo = copyCartInfo.filter(
+        (cartItem) => cartItem.id != payload.id
+      );
+
+      return {
+        count: state.count - parseInt(payload.quantity),
+        cartInfo: copyCartInfo,
+      };
```

Comment:
This way of creating `let` and updating in each case sentence is not recommendable, instead using `const` with a descriptive names. 

```suggestion
      const newCartState = state.cartInfo.filter(
        (cartItem) => cartItem.id != payload.id
      );

      return {
        count: state.count - parseInt(payload.quantity),
        cartInfo: newCartState,
      };
```

Reply:
done